docs/pages/enroll-resources/mcp-access/stdio.mdx

```diff
@@ -0,0 +1,170 @@
+---
+title: MCP Access with Stdio MCP Server
+description: How to access MCP server with stdio transport.
+---
+
+Teleport can provide secure access to MCP servers with stdio transport.
+
+This guides shows you how to:
+
+- Enroll a MCP server with stdio transport in your Teleport cluster.
+- Connect to the stdio MCP server via Teleport.
+
+## How it works
+
+![How it works](../../../img/mcp-access/architecture-stdio.svg)
+
+Users can configure their MCP clients such as Claude Desktop to start an MCP
+server using `tsh`. Once successfully authorized, `tsh` establishes a session
+with the Application Service.
+
+The Application Service starts the MCP server using the command and arguments
+defined by Teleport administrators in the app definition. Teleport then proxies
+the connection between the client and the remote MCP server, applying additional
+role-based access controls such as filtering which tools are available to the
+user. While proxying, Teleport also logs MCP protocol requests as audit events,
+providing visibility into user activity.
+
+## Prerequisites
+
+(!docs/pages/includes/edition-prereqs-tabs.mdx!)
+- A host, e.g., an EC2 instance, where you will run the Teleport Applications
+  Service.
+- Tools on the same host that can launch stdio-based MCP servers like `npx`
+  or `docker`.
+
+## Step 1/3. Configure the Teleport Application Service
+
+### Get a join token
+(!docs/pages/includes/tctl-token.mdx serviceName="Application" tokenType="app" tokenFile="/tmp/token"!)
+
+(!docs/pages/includes/database-access/alternative-methods-join.mdx!)
+
+### Install the Teleport Application Service
+Install Teleport on the host where you will run the Teleport Application Service:
+
+(!docs/pages/includes/install-linux.mdx!)
+
+### Configure the Teleport Application service
+
+On the host where you will run the Teleport Application Service, create a file
+at `/etc/teleport.yaml` with the following content:
+```yaml
+version: v3
+teleport:
+  join_params:
+    token_name: "/tmp/token"
+    method: token
+  proxy_server: "<Var name="teleport.example.com:443" />"
+auth_service:
+  enabled: false
+proxy_service:
+  enabled: false
+ssh_service:
+  enabled: false
+app_service:
+  enabled: true
+  apps:
+  - name: "everything"
+    labels:
+      env: dev
+    description:
+    mcp:
+      # Command to launch stdio-based MCP servers.
+      command: "docker"
+      # Args to execute with the command.
+      args: ["run", "-i", "--rm", "mcp/everything"]
+      # Name of the host user account under which the command will be
+      # executed. Required for stdio-based MCP servers.
+      run_as_host_user: "docker"
```

Comment:
Do we support registering MCP servers as yaml resources? `tctl create mcp.yaml`, same as regular apps, databases, etc. I think we should prefer using dynamic registration in guides rather than static config.

Reply:
Sure, we can do that.